gateway/gateway.go

```diff
@@ -466,7 +466,8 @@ func (gw *Gateway) SendMessage(
 
 	// exclude file delete event as the msg ID here is the native file ID that needs to be deleted
 	if msg.Event != config.EventFileDelete {
-		msg.ID = gw.getDestMsgID(rmsg.Protocol+" "+rmsg.ID, dest, channel)
+		canonicalMsgID := gw.FindCanonicalMsgID(rmsg.Protocol, rmsg.ID)
```

Comment:
to be honest this is a bit of a hacky workaround and I don't necessarily like doing this pretty expensive operation on basically every message recived.

Hypothetically we can work around this 

In practice this inefficiency would only matter in really large message caches (and we're mostly limited by I/O bandwidth).

I still don't really like it though.
I'd rather it be more of a graph-like structure where any message maps to the rest of the messages rather than 1 canonical message to the sent mesages.

Reply:
This is less of an issue with the persistent message maps in #1996